onnx/defs/generator/defs.cc

```diff
@@ -1,227 +1,292 @@
+    .Input(
+           0,
+           "input",
+           "Input tensor to copy shape and optionally type information from.", "T1")
+    .Output(
+            0,
+            "output",
+            "Output tensor of random values drawn from normal distribution", "T2")
+    .TypeConstraint(
+        "T1",
+        OpSchema::all_tensor_types(),
+        "Constrain to any tensor type. If the dtype attribute is not provided this must be a valid output type.")
+    .TypeConstraint("T2", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        if (ctx.getAttribute("dtype") != nullptr) 
+          propagateElemTypeFromAttributeToOutput(ctx, "dtype", 0);
+        else
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+    });
+
+ONNX_OPERATOR_SCHEMA(Multinomial)
+    .SetDoc(R"DOC(
+Generate a tensor of samples from a multinomial distribution according to the probabilities
+of each of the possible outcomes.
+)DOC")
+    .Attr(
+        "sample_size",
+        "Number of times to sample.",
+        AttributeProto::INT,
+        static_cast<int64_t>(1))
+    .Attr(
+        "seed",
+        "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+        AttributeProto::FLOAT,
+        OPTIONAL)
+    .Attr(
+        "dtype",
+        "(Optional) The data type for the elements of the output tensor, if not specified, we will use int32.",
+        AttributeProto::INT,
+        OPTIONAL)
+    .Input(
+        0,
+        "input",
+        "Input tensor with shape [batch_size, class_size], where class_size is the number of all possible outcomes. Each value along the axis zero represents the unnormalized log-probability of each corresponding outcome in a batch.", "T1")
+    .Output(
+        0,
+        "output",
+        "Output tensor with shape [batch_size, sample_size], where sample_size is the number of times to sample. Each value along the axis zero represents the outcome of the corresponding sample in a batch.", "T2")
+    .TypeConstraint(
+        "T1",
+        { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain input types to float tensors.")
+    .TypeConstraint(
+        "T2",
+        { "tensor(int32)", "tensor(int64)" },
+        "Constrain output types to integral tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        auto output_type = ctx.getOutputType(0);
+        auto input_type = ctx.getInputType(0);
+        if (input_type == nullptr || output_type == nullptr ||
+            TypeProto::kTensorType != output_type->value_case() ||
+            TypeProto::kTensorType != input_type->value_case()) {
+            return;
+        }
+        auto sample_size = ctx.getAttribute("sample_size");

```

Comment:
`sample_size` is an optional attribute and so here you can possibly get a `nullptr`

Reply:
fixed.

Reply:
@wchao1115 It is optional, you currently has set a default value for it, which defines how should the backend should interpret when it's missing. My previous comment was for required attribute.